public/app/features/storage/StoragePage.tsx

```diff
@@ -175,7 +263,20 @@ export default function StoragePage(props: Props) {
                 Dashboard
               </LinkButton>
             )}
-            {canAddFolder && <Button onClick={() => setIsAddingNewFolder(true)}>New Folder</Button>}
+
+            {canAddFolder && (
+              <>
+                <FileUpload
+                  accept={fileFormats}
+                  onFileUpload={onFileUpload}
+                  key={fileUploadKey}
+                  className={styles.uploadButton}
+                >
+                  Upload
+                </FileUpload>
+                <Button onClick={() => setIsAddingNewFolder(true)}>New Folder</Button>
+              </>
+            )}
```

Comment:
It might make the StoragePage component cleaner if we moved the `UploadButton` and related code to its own component (similar to how this was separated out into `UploadView` component before)

Reply:
I looked into that but I'm not sure if it's worth it since most of the checks need to be done (or are used) in `StoragePage`..

Reply:
If we move both the button and the confirm modal into a separate component together do we need the additional state in storage page component?

Reply:
Looks like we don't, I only kept the error messages.